protonj2-client/src/main/java/org/apache/qpid/protonj2/client/impl/ClientConnection.java

```diff
@@ -886,10 +886,11 @@ private void initializeProtonResources(ReconnectLocation location) throws Client
             protonConnection.setContainerId(connectionId);
         }
 
+        final String hostname = (options.virtualHost() == null || options.virtualHost().length() < 1) ? location.getHost() : options.virtualHost();
```

Comment:
nit: I'd go with `options.virtualHost().isEmpty()` for the second part of the condition.

Reply:
Code updated as suggested.